OptionSelector/OSCategoryTBV.m

```diff
@@ -0,0 +1,177 @@
+                       @"Comedy pals",
+                       @"Tony",
+                       @"C4Q friends + Mike",
+                       @"Roommates",
+                       @"Mi hermano",
+                ];
+    buddies.selection = @" ";   // leave selection empty
+    
+    [self.allCategories addObject:buddies]; //adds buddies into our NSMutableArray
+    
+    CQCategory * food = [[CQCategory alloc]init];
+    
+    food.name = @"What to eat";
+    food.options = @[
+                     @"Chipotle",
+                     @"Pizza",
+                     @"Indian food",
+                     @"Bahn Mi & Thai Iced Tea",
+                     @"Lobster rolls",
+                     @"Wings & beer",
+                     @"Doughnut Plant donuts",
+                     @"Go Go Curry"
+                ];
+    food.selection = @" ";
+    
+    [self.allCategories addObject:food];
+    
+    CQCategory * activity = [[CQCategory alloc]init];
+    
+    activity.name = @"Where we hanging out?";
+    activity.options = @[
+                        @"The Spot",
+                        @"Beergarden",
+                        @"My apartment",
+                        @"At C4Q",
+                        @"Rockaway beach",
+                        @"Karaoke Cave",
+                        @"Cabin in Woodstock",
+                        @"Highline",
+                        @"Walter's Bar",
+                        @"Columbia campus"
+                         ];
+    activity.selection = @" ";
+    
+    [self.allCategories addObject:activity];
+    
+    
+ 
+}
+
+- (void) viewWillAppear:(BOOL)animated {
+    [self.tableView reloadData];                //reloads data to show the selection in cell detail
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return self.allCategories.count; //returning the count of all objects in allCategories array
+}
+
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"categoryReuseIdentifier" forIndexPath:indexPath];
+    
+    CQCategory *thisCategory = [self.allCategories objectAtIndex:indexPath.row];  //taking the category instance and putting it in variable of this category. use the class as a variable and not string
+    
+    cell.textLabel.text = thisCategory.name; //uses name property as label of string
+   
+        if (thisCategory.selection.length == 0){
+            cell.detailTextLabel.text = @" ";
+        } else {
+            cell.detailTextLabel.text = thisCategory.selection;
+        }
+    return cell;
+    
+    
+}
+
+- (void) prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    
+    if ([[segue identifier] isEqualToString:@"showOptionsSegue"]){
```

Comment:
Since there are no other Segues in this view controller, you can probably get away without this if condition.